core/stop/residual_norm.cpp

```diff
@@ -234,6 +236,212 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_RESIDUAL_NORM);
     class ImplicitResidualNorm<_type>
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_IMPLICIT_RESIDUAL_NORM);
 
+class ResidualNormFactory;
+
+struct residual_norm_factory_parameters
+    : public enable_parameters_type<residual_norm_factory_parameters,
+                                    ResidualNormFactory> {
+    double GKO_FACTORY_PARAMETER_SCALAR(threshold, 0.0);
```

Comment:
nit: maybe call this `reduction_factor` to be consistent with the original residual norm parameters.